.github/workflows/report.yml

```
@@ -0,0 +1,55 @@
+name: Generate Report
+
+on:
+  workflow_call:
+
+jobs:
+  generate-report:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: zulu
+          java-version: 21
+      - name: Download allure results
+        id: download-allure-results
+        uses: actions/download-artifact@v4
+        with:
+          pattern: allure_results_*
+          merge-multiple: true
+          path: allure-results
+
+      - name: Load test report history
+        id: allure-history
+        uses: actions/checkout@v3
+        continue-on-error: true # in case run without branch for GitHub page source
+        with:
+          ref: gh-pages
+          path: gh-pages
+
+      - name: Build test report
+        id: allure-report
+        if: always()
+        uses: simple-elf/allure-report-action@v1.8
+        with:
+          gh_pages: gh-pages
+          allure_history: allure-history
+          allure_results: allure-results
+          allure_report: allure-report
+          keep_reports: 10
+
+      - name: Publish test report
+        id: deploy-allure-report-page
+        if: steps.allure-report.outcome == 'success' # make sure the allure report is generated successfully
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_branch: gh-pages
+          publish_dir: allure-history
+
+      - name: Create job summaries
+        if: steps.deploy-allure-report-page.outcome == 'success'
+        run: |
+          pages_url=https://$(echo $GITHUB_REPOSITORY | awk -F '/' '{print $1}').github.io/$(echo $GITHUB_REPOSITORY | awk -F '/' '{print $2}')/${{ github.run_number }}/#behaviors
```

We can simplify as the following.

```suggestion
          pages_url=https://${GITHUB_REPOSITORY_OWNER}.github.io/${GITHUB_REPOSITORY#${GITHUB_REPOSITORY_OWNER}/}/${{ github.run_number }}/#behaviors
```

---

`${{ github.run_number }}` won't change in retrying.
How about adding `github.run_attempt` as the suffix?

> This number does not change if you re-run the workflow run.

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

---

With `github.run_attempt`, It returns the number of attempt (start from 1), not the run number.
But, the generated github page is based on the run number.

---

Do you mean it is an expected behavior that rerunning the test run will override the previous attempt?
If so, there is no problem!

---

Oh sorry for the confusion.
I mean if we use `github.run_attempt`, the output report link to github pages always starts from 1.
- For example, `https://kintone.github.io/cli-kintone/1/#behaviors`
- When re-run, It's `https://kintone.github.io/cli-kintone/2/#behaviors`

But, the actual generated report page on github page is created with the run number. (Currently, It's `4589`)
So, the link should be `https://kintone.github.io/cli-kintone/4589/#behaviors`

For the next workflow run, It should be `https://kintone.github.io/cli-kintone/4590/#behaviors`.
==========================

I think when re-run the workflow, It's not necessary to keep multiple versions of the report. We just need the latest report for each workflow.

---

I meant to apply `github.run_attempt` as the suffix, e.g. 4589-1, 4589-2, ...
(It's a common way to distinguish reruns)

OK, it seems fine not to keep multiple versions.

---

@tasshi-me 

> OK, it seems fine not to keep multiple versions.
Okay.

> e.g. ${{ github.run_number }}-${{ github.run_attempt }} -> 4589-1, 4589-2, ...
From my understanding, we can not control the page ID of the github page when it's created. The page ID will be created based on the current run number of the workflow, such as `4589`, `4590`

At step `Create job summaries`, we just create a link to the github page based on that run number.
So, If we create a link like `https://kintone.github.io/cli-kintone/4589-1/`, It will redirect to 404 page.

---

If we don't keep multiple versions, we don't need to use `${{ github.run_attempt }}`.
If we want to keep multiple versions, we have to update both `Create job summaries` and link.

We decided we don't keep multiple versions, so you can go without `${{ github.run_attempt }}` now.